### PR TITLE
Add angular guard/yeremenko

### PR DESCRIPTION
**Pull request by Sergey988**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -21,12 +22,26 @@ import { WorkspaceCalendarComponent } from './presentation/components/workspace-
 import { TwoSetpVerificationComponent } from './presentation/components/user-components/two-step-verification/two-step-verification.component';
 
 const userWorkspaceRouters: Routes = [
-  { path: 'managment', component: WorkspaceMemberManagmentComponent, pathMatch: 'full'},
+  {
+    path: 'managment',
+    component: WorkspaceMemberManagmentComponent,
+    canActivate: [PermissionGuard],
+    data:{
+      userRoles: [1,2]
```
Comment (YevheniiVovk) [inline src/app/app-routing.module.ts:30]: Create tech dept for adding permissions on server side and then integrate it here. Use hard code is a bad solution.

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -21,12 +22,26 @@ import { WorkspaceCalendarComponent } from './presentation/components/workspace-
...
   { path: 'calendar', component: WorkspaceCalendarComponent},
   { path: 'info', component: WorkspaceInfoComponent},
   { path: 'tasklist', component: WorkspaceTaskListComponent},
-  { path: 'settings', component: WorkspaceSettingsComponent}
+  {
+    path: 'settings',
+    component: WorkspaceSettingsComponent,
+    canActivate: [PermissionGuard],
+    data:{
+      userRoles: [1]
```
Comment (YevheniiVovk) [inline src/app/app-routing.module.ts:42]: Same here

File: src/app/core/guards/permission.guard.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  providedIn: 'root'
+})
+
+export class PermissionGuard implements CanActivate {
+
+  currentUserRole: number;
+  workspaceId: number;
+  allowedUserRoles: number[];
+
+  constructor(private router: Router, private workspaceService: WorkspaceService, private activeRoute: ActivatedRoute){}
```
Comment (YevheniiVovk) [inline src/app/core/guards/permission.guard.ts]: more then 100 ch in line

File: src/app/core/guards/permission.guard.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  async canActivate(route: ActivatedRouteSnapshot) : Promise<boolean>{
+
+    if(route.routeConfig?.data){
+      this.allowedUserRoles = route.routeConfig.data['userRoles'];
+      this.workspaceId = route.parent?.params['id'];
+      this.currentUserRole = (await firstValueFrom(this.workspaceService.getWorkspaceInfo(this.workspaceId))).role;
```
Comment (YevheniiVovk) [inline src/app/core/guards/permission.guard.ts]: more then 100 ch in line

File: src/app/core/guards/permission.guard.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+      if(this.allowedUserRoles.indexOf(this.currentUserRole) != -1) {
+        return true;
+      }
+
+      this.showErrorAlert('you have no permission to view this page');
```
Comment (YevheniiVovk) [inline src/app/core/guards/permission.guard.ts]: Should we move static  messages into some file

File: src/app/core/guards/permission.guard.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      return false;
+    }
+    return true;
+  }
+
+  showErrorAlert(error: string){
+    Swal.fire({
+      position: 'center',
+      icon: 'error',
+      title: 'No permission',
```
Comment (YevheniiVovk) [inline src/app/core/guards/permission.guard.ts]: Same here

File: src/app/presentation/components/workspace-components/workspace-menu/workspace-menu.component.html [hunk 7]
```diff
@@ -9,16 +9,16 @@
     <ul>
       <li>
         <ul class="group-menu">
-          <li class="menu-item" *ngIf="userWorkspaceInfo.role == 1 || userWorkspaceInfo.role == 2" >
+          <li *appIfUserRoles='[1,2]' class="menu-item">
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/workspace-menu/workspace-menu.component.html:12]: Create tech dept for adding permissions on server side and then integrate it here. Use hard code is a bad solution.

File: src/app/presentation/components/workspace-components/workspace-menu/workspace-menu.component.html [hunk 8]
```diff
@@ -28,46 +28,46 @@
...
         </li>
         <li class="menu-item">
           <a [routerLink]="'/user/workspace/' +userWorkspaceInfo.id + '/info'">
             <mat-icon aria-hidden="false" aria-label="Example home icon">info</mat-icon>
-            <span>Workspace info</span>  
+            <span>Workspace info</span>
           </a>
         </li>
-        <li class="menu-item">
+        <li *appIfUserRoles='[1]' class="menu-item">
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/workspace-menu/workspace-menu.component.html:67]: Same here

**Review by YevheniiVovk — APPROVED** [review]

**Review by FeloniukM — APPROVED** [review]

**Pull request merged**